packages/build-tools/src/utils/outputs.ts

```diff
@@ -0,0 +1,80 @@
+import { Generic } from '@expo/eas-build-job';
+import { BuildStep, jsepEval } from '@expo/steps';
+import { bunyan } from '@expo/logger';
+import nullthrows from 'nullthrows';
+import fetch from 'node-fetch';
+
+import { BuildContext } from '../context';
+
+export async function uploadJobOutputsToWwwAsync(
+  ctx: BuildContext<Generic.Job>,
+  {
+    steps,
+    logger,
+    expoApiV2BaseUrl,
+  }: { steps: BuildStep[]; logger: bunyan; expoApiV2BaseUrl: string }
+): Promise<void> {
+  if (!ctx.job.outputs) {
+    logger.info('Job defines no outputs, skipping upload');
+    return;
+  }
+
+  try {
+    const workflowJobId = nullthrows(ctx.job.builderEnvironment?.env?.__WORKFLOW_JOB_ID);
```

Comment:
If the user maliciously changes the `__WORKFLOW_JOB_ID` value to some other workflow ID they won't be able to write outputs values for it because we validate it in WWW, right?

BTW It would be good to move it to `job.workflowId` so it can't be overridden by the user (not a part of this PR)

Reply:
https://linear.app/expo/issue/ENG-14365/move-builderenvironmentenv-workflow-id-to-jobworkflowid